lib/_http_agent.js

```diff
@@ -496,7 +496,10 @@ Agent.prototype.keepSocketAlive = function keepSocketAlive(socket) {
       const hint = RegExpPrototypeExec(/^timeout=(\d+)/, keepAliveHint)?.[1];
 
       if (hint) {
-        const serverHintTimeout = NumberParseInt(hint) * 1000;
+        // Let the timer expires before the announced timeout to reduce
+        // the likelihood of ECONNRESET errors
+        let serverHintTimeout = ( NumberParseInt(hint) * 1000 ) - 1000;
```

Comment:
In the unlikely case the server responds with a 1 second keepalive, this will set the disable the timeout because serverHintTimeout will be 0 -- maybe go with `- 500` in place of `- 1000` ?